### ISSUE TITLE
Kingdom Hearts 1.5 + 2.5 Kingdom Hearts 2 Final Mix "Archive Contains a hash which is not in Filenames table" 

### ISSUE THREAD

**Issue by Kite2810** [rm-NoobInCoding/UnPSARC#21]
While extracting the psarc I got this issue. It still extracts all the files as .bin ofcourse.

**Comment by Kite2810** [comment]
Ah idk if that helps... but I have a hashlist for the game. The one will be used for PS2 or PC version of the game.
[_HashPairs PS2.log](https://github.com/user-attachments/files/16830881/_HashPairs.PS2.log)
[kh2idxPC.txt](https://github.com/user-attachments/files/16830882/kh2idxPC.txt)

**Comment by rm-NoobInCoding** [comment]
Please test v2.7 and see if its fixed or not

**Comment by Kite2810** [comment]
Yep for kh2_first.psarc it works! it just seems the files are encrypted.
kh2_second.psarc gives now this error tho: UnPSARC (Archive Tool For PlayStation Archive files 'PSARC')
By NoobInCoding
https://github.com/rm-NoobInCoding/UnPSARC

Unpacking kh2_second.psarc...
Files count: 0

Unbehandelte Ausnahme: System.ArgumentException: Ein Element mit dem gleichen Schlüssel wurde bereits hinzugefügt.
   bei System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   bei System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   bei UnPSARC.PSARC.LoadFileNames(Byte[] file)
   bei UnPSARC.PSARC.Read()
   bei UnPSARC.Archive.Unpack(Stream ArchiveRaw, String Folder)
   bei UnPSARC.Program.UnpackArchiveFile(String inputPath, String outputDirectory)
   bei UnPSARC.Program.Main(String[] args)

**Comment by rm-NoobInCoding** [comment]
Please upload the file

**Comment by Kite2810** [comment]
here :) 
https://drive.google.com/file/d/1R2iviUTxWFaBT2f7e0Hl6y4amPe_Ecx6/view?usp=drive_link
https://drive.google.com/file/d/1gT8GNB79u0Atey2T-9ikfA-qhYGCFCkq/view?usp=drive_link

**Comment by rm-NoobInCoding** [comment]
I just updated the v2.7 release. test it.
